src/abort.rs

```diff
@@ -64,3 +59,35 @@ impl<I, V, E> IteratorExt<I, V, E> for I
     }
 }
 
+impl<I, V, IE, OE> IteratorExt<I, V, IE> for Result<I, OE>
+    where I: Iterator<Item = Result<V, IE>> + Sized,
+          OE: Debug
+{
+    fn abort_on_err(self) -> AbortingIter<I, V, IE> {
+        AbortingIter::from(self.unwrap_or_abort())
+    }
+}
+
+
+/// Extension trait for convenient abortion in case of errors
+///
+pub trait Abortable<V>
+{
+    /// Just like a regulat unwrap() except it performs proper logging
+    ///
+    /// Returns the contained value or aborts the program, logging the error.
+    ///
+    fn unwrap_or_abort(self) -> V;
+}
+
+impl<V, E> Abortable<V> for Result<V, E>
+    where E: Debug
```

Comment:
The `ErrorChain` trait implements `Display`. The documentation states that the entire chain will be printed in such cases. I assumed that `error!()` will do the right thing in such cases.